File: py/portfolio.py
import functools
import operator
from money import Money

class Portfolio:
    def __init__(self):
        self.moneys = []
        
    def __convert(self, money, currency):
        exchangeRates = {"EUR->USD": 1.2, "USD->KRW": 1100}
        value = 1
        if money.currency == currency:
            value = money.amount
        else:
            key = "%s->%s" % (money.currency, currency)
            value = money.amount * exchangeRates[key]
        return value
    
    def add(self, *moneys):
        self.moneys.extend(moneys)
    
    def evaluate(self, currency):
        total = functools.reduce(
            operator.add, 
            map(lambda m: self.__convert(m, currency), self.moneys), 
            0
        )
        return Money(total, currency)

File: js/portfolio.js
import Money from './money.js';

export default class Portfolio {
    constructor() {
        this.moneys = [];
    }

    add(...moneys) {
        this.moneys = this.moneys.concat(moneys);
    }

    convert(money, currency) {
        // const eurToUsd = 1.2;
        const exchangeRates = new Map();
        exchangeRates.set('EUR->USD', 1.2);
        exchangeRates.set('USD->KRW', 1100);

        let value = 1;

        if (money.currency === currency) {
            value = money.amount;
        }
        else {
            let key = `${money.currency}->${currency}`;
            let rate = exchangeRates.get(key);
            value = money.amount * rate;
        }

        return value;
    }

    evaluate(currency) {
        let total = this.moneys.reduce((sum, money) => {
            return sum + this.convert(money, currency);
        }, 0);
        return new Money(total, currency);
    }
}